OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnHealth.cs

```diff
@@ -0,0 +1,99 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Applies a condition to the actor at when its health is between 2 specific values.")]
+	public class GrantConditionOnHealthInfo : ITraitInfo, IRulesetLoaded, Requires<IHealthInfo>
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		[Desc("Play a random sound from this list when enabled.")]
+		public readonly string[] EnabledSounds = { };
+
+		[Desc("Play a random sound from this list when disabled.")]
+		public readonly string[] DisabledSounds = { };
+
+		[Desc("Minimum level of health at which to grant the condition.")]
+		public readonly int MinHP = 0;
+
+		[Desc("Maximum level of health at which to grant the condition.")]
```

Comment:
I think this should mention that a value of `0` ( `-1` might be a better value) will cause a fallback to Health.MaxHP.